src/daft-local-execution/src/runtime_stats/subscribers/progress_bar.rs

```diff
@@ -201,27 +203,38 @@ impl RuntimeStatsSubscriber for IndicatifProgressBarManager {
     }
 
     async fn finish(mut self: Box<Self>) -> DaftResult<()> {
-        self.pbars.clear();
-        self.multi_progress.clear()?;
+        if !self.persist_on_finish {
+            self.pbars.clear();
+            self.multi_progress.clear()?;
+        }
```

Comment:
Just to sanity check, can you check if we need to call `multi_progress.finish()` in the Indicatif library (https://docs.rs/indicatif/latest/indicatif/). I know that sometimes the progress bar clobbers over other prints, and I believe `.finish()` will help avoid that?